libs/pageserver_api/src/models.rs

```diff
@@ -411,12 +411,16 @@ pub struct LayerResidenceEvent {
     pub reason: LayerResidenceEventReason,
 }
 
-/// The reason for recording a given [`ResidenceEvent`].
+/// The reason for recording a given [`LayerResidenceEvent`].
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub enum LayerResidenceEventReason {
     /// The layer map is being populated, e.g. during timeline load or attach.
     /// This includes [`RemoteLayer`] objects created in [`reconcile_with_remote`].
     /// We need to record such events because there is no persistent storage for the events.
+    ///
+    // https://github.com/rust-lang/rust/issues/74481
+    /// [`RemoteLayer`]: ../pageserver/tenant/storage_layer/struct.RemoteLayer.html
+    /// [`reconcile_with_remote`]: ../pageserver/tenant/struct.Timeline.html#method.reconcile_with_remote
```

Comment:
Is the rust issue meant to be linekd here?

Reply:
The rust issue is just that we have to link to the html because intra doc links don't support linking to dependents.

But the links actually don't work, I need to fix them, thanks for bringing this up.